sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -148,6 +149,8 @@ private[sql] class SharedState(val sparkContext: SparkContext) extends Logging {
 
 object SharedState {
 
+  URL.setURLStreamHandlerFactory(new SparkUrlStreamHandlerFactory())
```

Comment:
Say one day we need to support a third-party file system named "shaofs", who provides a ShaoUrlStreamHandlerFactory. We cannot set both FsUrlStreamHandlerFactory and ShaoUrlStreamHandlerFactory to JVM, but we can set SparkUrlStreamHandlerFactory which deliver commands to ShaoUrlStreamHandlerFactory and FsUrlStreamHandlerFactory accordingly.